steely/plugins/btc.py

```diff
@@ -1,33 +1,64 @@
 '''Get the latest Bitcoin price (in USD and EUR)'''
 import requests
 
-__author__ = 'CianLR'
+__author__ = ('CianLR', 'byxor')
 COMMAND = 'btc'
 
 API_URL = 'https://api.coindesk.com/v1/bpi/currentprice.json'
 RESP_TEMPLATE = """Bitcoin Price:
-    €{}
-    ${}"""
+€{}
+${}
+Difference since last time: {}"""
 
-def is_valid_response(resp):
-    return ('bpi' in resp and
-            'EUR' in resp['bpi'] and 'rate' in resp['bpi']['EUR'] and
-            'USD' in resp['bpi'] and 'rate' in resp['bpi']['USD'])
+last_euros = 100
+
+class BitcoinException(Exception): pass
+class BadHttpResponse(BitcoinException): pass
+class MissingFields(BitcoinException): pass
+
+
+def get_bitcoin_rates():
+    response = requests.get(API_URL)    
 
-def get_btc():
-    response = requests.get(API_URL)
     if not response.ok:
-        return "Error getting price. Response (code {}) {}".format(
+        message = "Error getting price. Response (code {}) {}".format(
             response.status_code, response.text)
+        raise BadHttpResponse(message)
+    
     j = response.json()
-    if not is_valid_response(j):
-        return "Response is missing fields: {}".format(str(j))
-    return RESP_TEMPLATE.format(j['bpi']['EUR']['rate'],
-                                j['bpi']['USD']['rate'])
+    
+    if not _is_valid_response(j):
+        message = "Response is missing fields: {}".format(str(j))
+        raise MissingFields(message)
+    
+    euros = float(j['bpi']['EUR']['rate'])
+    dollars = float(j['bpi']['USD']['rate'])
+    return (euros, dollars)
 
-def main(bot, author_id, message, thread_id, thread_type, **kwargs):
-    bot.sendMessage(get_btc(), thread_id=thread_id, thread_type=thread_type)
+    
+def _is_valid_response(resp):
+    return ('bpi' in resp and
+            'EUR' in resp['bpi'] and 'rate' in resp['bpi']['EUR'] and
+            'USD' in resp['bpi'] and 'rate' in resp['bpi']['USD'])
+
+
+def percentage_increase(before, after):
+    delta = after - before
+    return (delta * 100) / before
 
-if __name__ == '__main__':
-    print(get_btc())
 
+def percentage_string(increase):
+    prefix = "+" if increase >= 0 else ""
```

Comment:
is there not a format modifier for this behavior directly?

Reply:
Implemented.